purge/purge.py

```diff
@@ -352,9 +359,48 @@ def wait_and_purge_room(room_id=None, event_id=None):
         if docker_restart_label:
             client = docker.from_env()
             for container in client.containers.list():
-                if container.attrs['State']['Status'] == 'running' and\
-                        container.attrs['Config']['Labels'].get(docker_restart_label):
-                    container.restart(timeout=30)
+                if container.attrs['State']['Status'] != 'running' or\
+                        not container.attrs['Config']['Labels'].get(docker_restart_label):
+                    continue
+
+                try:
+                    # parse container's env vars
+                    env_vars = dict(
+                        itemgetter(0, 2)(e.partition('='))
+                        for e in container.attrs['Config']['Env']
+                    )
+                    remote_config_file = env_vars.get(
+                        'URL_KNOWN_FEDERATION_SERVERS',
+                    ) or URL_KNOWN_FEDERATION_SERVERS_DEFAULT
+
+                    # fetch remote file
+                    remote_whitelist = yaml.load(requests.get(remote_config_file).text)
```

Comment:
You're making a request per container to the same resource... could be done once outside the loop

Reply:
@rakanalh there should always be only one container which matches the filter (`docker_restart_label` set), therefore request was made under for loop to simplify `try/except` clause.

Reply:
Still, if you end up for some reason with 2 containers then the request is going to happen twice.

Reply:
Actually, looking at the code I remembered, I need to do it per-container, as different containers could have different `URL_KNOWN_FEDERATION_SERVERS` environment variables